File: dotnet/dotnet_b2b_seeding/dotnet_service/Database/Models/ClientApp.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotnetService.Database.Models
{
    /// <summary>
    /// ClientApp model.
    /// Similar to the Python client_app.py implementation.
    /// </summary>
    [Table("client_apps")]
    public class ClientApp : BaseModel
    {
        [Required]
        [MaxLength(100)]
        [Column("name")]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(50)]
        [Column("code")]
        public string Code { get; set; } = string.Empty;
        
        [MaxLength(100)]
        [Column("apiKey")]
        public string? ApiKey { get; set; }
        
        [Required]
        [Column("description", TypeName = "text")]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        [Column("ownerUserId")]
        public Guid OwnerUserId { get; set; }
        
        [Column("organizationId")]
        public Guid? OrganizationId { get; set; }
        
        [MaxLength(500)]
        [Column("redirectUri")]
        public string? RedirectUri { get; set; }
        
        [MaxLength(500)]
        [Column("logoUrl")]
        public string? LogoUrl { get; set; }
        
        [MaxLength(500)]
        [Column("websiteUrl")]
        public string? WebsiteUrl { get; set; }
        
        [MaxLength(500)]
        [Column("privacyPolicyUrl")]
        public string? PrivacyPolicyUrl { get; set; }
        
        [MaxLength(500)]
        [Column("termsOfServiceUrl")]
        public string? TermsOfServiceUrl { get; set; }
        
        [Column("isVerified")]
        public bool IsVerified { get; set; } = false;
        
        [Column("isActive")]
        public bool IsActive { get; set; } = true;
        
        [Required]
        [MaxLength(20)]
        [Column("status")]
        public string Status { get; set; } = "active";
        
        [Column("deletedAt")]
        public DateTime? DeletedAt { get; set; }
    }
}

File: python/python_b2b_seeding/python_service/database/models/client_app.py
"""
ClientApp model.
Similar to the Node.js client-app.model.ts implementation.
"""
from sqlalchemy import Column, String, Boolean, Text, DateTime
from database.models.base import BaseModel
from database.models.types import GUID
import uuid


class ClientApp(BaseModel):
    """ClientApp model."""
    __tablename__ = 'client_apps'
    
    # Explicitly define id first to ensure it appears first in database
    id = Column(GUID(), primary_key=True, default=uuid.uuid4)
    
    name = Column(String(100), nullable=False)
    code = Column(String(50), unique=True, nullable=False, index=True)
    api_key = Column(String(100), name='apiKey', nullable=True)
    description = Column(Text, nullable=False)
    owner_user_id = Column(GUID(), name='ownerUserId', nullable=False, index=True)
    organization_id = Column(GUID(), name='organizationId', nullable=True, index=True)
    
    redirect_uri = Column(String(500), name='redirectUri', nullable=True)
    logo_url = Column(String(500), name='logoUrl', nullable=True)
    website_url = Column(String(500), name='websiteUrl', nullable=True)
    privacy_policy_url = Column(String(500), name='privacyPolicyUrl', nullable=True)
    terms_of_service_url = Column(String(500), name='termsOfServiceUrl', nullable=True)
    
    is_verified = Column(Boolean, name='isVerified', default=False, nullable=False)
    is_active = Column(Boolean, name='isActive', default=True, nullable=False)
    status = Column(String(20), default='active', nullable=False)
    
    deleted_at = Column(DateTime, name='deletedAt', nullable=True)

